994_Rotting_Oranges.py

```diff
@@ -0,0 +1,48 @@
+# Time: O(n**2)
+# Space: O(1)
+
+
+class Solution(object):
+    def orangesRotting(self, grid):
+        """
+        :type grid: List[List[int]]
+        :rtype: int
+        """
+        row, col = len(grid), len(grid[0])
+        visited = set()
+        rotted_group = []
+        ans = -1
+
+        for row_num in range(row):
+            for col_num in range(col):
+                if grid[row_num][col_num] == 2:
+                    rotted_group.append((row_num, col_num))
+
+        while rotted_group:
+            for x in range(len(rotted_group)):
+                row_num, col_num = rotted_group.pop(0)
+                if (row_num, col_num) in visited:
+                    continue
+                visited.add((row_num, col_num))
+
+                if row_num + 1 < row and grid[row_num+1][col_num] == 1:
+                    grid[row_num+1][col_num] = 2
+                    rotted_group.append((row_num + 1, col_num))
+                if row_num - 1 >= 0 and grid[row_num - 1][col_num] == 1:
+                    grid[row_num-1][col_num] = 2
+                    rotted_group.append((row_num-1, col_num))
+                if col_num + 1 < col and grid[row_num][col_num + 1] == 1:
+                    grid[row_num][col_num+1] = 2
+                    rotted_group.append((row_num, col_num + 1))
+                if col_num - 1 >= 0 and grid[row_num][col_num - 1] == 1:
+                    grid[row_num][col_num - 1] = 2
+                    rotted_group.append((row_num, col_num - 1))
```

Comment:
You can compress these `if` blocks into one `for` loop. Refer to [this code](https://github.com/okuda-seminar/leetcode_answers/blob/main/994_rotting_oranges.py).